tests/countries/test_north_korea.py

```diff
@@ -0,0 +1,253 @@
+#  holidays
+#  --------
+#  A fast, efficient Python library for generating country, province and state
+#  specific sets of holidays on the fly. It aims to make determining whether a
+#  specific date is a holiday as fast and flexible as possible.
+#
+#  Authors: Vacanza Team and individual contributors (see CONTRIBUTORS file)
+#           dr-prodigy <dr.prodigy.github@gmail.com> (c) 2017-2023
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/vacanza/holidays
+#  License: MIT (see LICENSE file)
+
+from unittest import TestCase
+
+from holidays.countries.north_korea import NorthKorea, KP, PRK
+from tests.common import CommonCountryTests
+
+
+class TestNorthKorea(CommonCountryTests, TestCase):
+    @classmethod
+    def setUpClass(cls):
+        years = range(1948, 2050)
+        super().setUpClass(NorthKorea, years=years)
+
+    def test_country_aliases(self):
+        self.assertAliases(NorthKorea, KP, PRK)
+
+    def test_no_holidays(self):
+        self.assertNoHolidays(NorthKorea(years=1947))
+
+    def test_new_years_day(self):
+        self.assertHolidayName("양력설", (f"{year}-01-01" for year in range(1948, 2050)))
+
+    def test_korean_new_years_day(self):
+        name = "설명절"
+        self.assertHolidayName(
+            name,
+            "2020-01-25",
+            "2021-02-12",
+            "2022-02-01",
+            "2023-01-22",
+            "2024-02-10",
+            "2025-01-29",
+        )
+        self.assertNoHolidayName(name, range(1968, 1989))
```

Comment:
For movable holidays, we should assert as well if they exist in all relevant years:
```suggestion
        )
        self.assertHolidayName(name, range(1948, 1968), range(1989, 2050))
        self.assertNoHolidayName(name, range(1968, 1989))
```